src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -798,17 +864,243 @@ def __try_add_error(error_list, detail):
         return {
-            "code": Constants.PatchOperationTopLevelErrorCode.SUCCESS if error_count_by_operation == 0 else Constants.PatchOperationTopLevelErrorCode.ERROR,
+            "code": code,
             "details": errors_by_operation,
             "message": message
         }
+
+    def __set_error_detail(self, error_code, message):
+        formatted_message = self.__ensure_error_message_restriction_compliance(message)
+        return {
+            "code": str(error_code),
+            "message": str(formatted_message)
+        }
     # endregion
 
+    # region - Patch Truncation
+    def log_truncated_packages(self):
+        """ log the removed packages from patches in CoreMain after main operation are marked completed """
+        if not len(self.__assessment_packages_removed) == 0:
+            self.composite_logger.log_debug("Packages removed from assessment packages list: {0}".format(self.__assessment_packages_removed))
+        if not len(self.__installation_packages_removed) == 0:
+            self.composite_logger.log_debug("Packages removed from installation packages list: {0}".format(self.__installation_packages_removed))
+        if len(self.__assessment_packages_removed) == 0 and len(self.__installation_packages_removed) == 0:
+            self.composite_logger.log_debug("No packages truncated")
+
+    def __check_truncation_time_passed_by_x_sec(self, track_truncation_timestamp):
+        """ check current sys time is more than truncation time stamp by x constant seconds"""
+        if track_truncation_timestamp is None:
+            track_truncation_timestamp = datetime.datetime.now()
+
+        curr_timestamp = datetime.datetime.now()
+        return (curr_timestamp - track_truncation_timestamp).total_seconds() >= Constants.StatusTruncationConfig.SKIP_TRUNCATION_LOGIC_IN_X_SEC
+
+    def __set_force_truncation_true_for_terminal_status(self, status):
+        """ status file needs truncation and last operation is either success or error, then force truncation on last operation"""
+        if self.__is_file_truncated and (status == Constants.STATUS_SUCCESS or status == Constants.STATUS_ERROR):
+            self.__force_truncation_on = True
+
+    def __check_status_file_size_perform_truncation_if_needed(self, status_file_payload_json_dumps):
+        status_file_size_in_bytes = self.__calc_status_size_on_disk(status_file_payload_json_dumps)  # calc complete_status_file_payload byte size on disk
+
+        if status_file_size_in_bytes > self.__internal_file_capacity:  # perform truncation complete_status_file byte size > 126kb
+            is_delay_truncation_by_x_sec = self.__check_truncation_time_passed_by_x_sec(self.__track_truncation_timestamp)
+
+            if is_delay_truncation_by_x_sec or self.__force_truncation_on:
```

Comment:
Should this condition read as:
If there is no delay required for truncation OR force truncation is on?

If so, the check should be:
 if not is_delay_truncation_by_x_sec or self.__force_truncation_on:

This is why a description conveying the intent/ desired functionality of the code is important, either in function description or PR description

Also, the function and variable names need to be modified to reflect the intent, but leaving that for the next review